src/main/java/user_use_case/gateways/UserGateway.java

```diff
@@ -204,17 +204,83 @@ public User getUser(String username) {
         return null;
     }
 
-    //@Override
-    //public void removeUser(User user) {
-        // DOES NOT WORK CURRENTLY
+    @Override
+    public void removeUser(User user) {
```

Comment:
You should just take in the username String since the other information of the User is irrelevant